Add example and usecase using Partition by (#5677)
Codecademy/docs: content/sql/concepts/window-functions/terms/lag/lag.md

| jenna      | black     | 35  | 60           |
| chris      | smith     | 30  | 35           |
| dave       | james     | 19  | 30           |

> ## Example using partition by

Reviewer: ```suggestion
## Example 
```
Remove this `using partition by` from the example heading, and you can place it below the ##Example heading. 
For example, like this below:
##Example
###using partition by
//Example code

This example demonstrates how to use the `LAG()` function to create a new column, `previous_position`. The function retrieves the value from the previous row within the same `employee_id` group in the `Promotions` table. This is achieved by using the `PARTITION BY employee_id` clause.

> Promotions Table

Reviewer: ```suggestion
`Promotions` Table
```

| employee_id | promotion_date | new_position |
| ----------- | -------------- | ------------ |
| 1      	  | 2020-01-01 	   | Junior Dev   |
| 1      	  | 2021-06-01 	   |    Mid Dev	  |
| 1      	  | 2024-03-01 	   | Senior Dev   |
| 2      	  | 2019-05-01     |     Intern	  |
| 2      	  | 2022-11-01 	   |    Analyst   |
| 2      	  | 2024-11-20     |Data Analyst  |

```sql
SELECT *,
    LAG(new_position) OVER (
        PARTITION BY employee_id
        ORDER BY promotion_date
    ) AS previous_position
FROM Promotions;
```

The output is a table that features a new column `previous_position`, which holds the values by employee. The first record is null because a default was not specified and the previous row would be out of range.

> Output
> 
> | employee_id | promotion_date | new_position | previous_position |
> | ----------- | -------------- | ------------ | ----------------- |
> | 1      	  | 2020-01-01 	   | Junior Dev   |              NULL |
> | 1      	  | 2021-06-01 	   |    Mid Dev	  | 	   Junior Dev |
> | 1      	  | 2024-03-01 	   | Senior Dev   |		      Mid Dev |
> | 2      	  | 2019-05-01     |     Intern	  |				 NULL |
> | 2      	  | 2022-11-01 	   |    Analyst   |			   Intern |
> | 2      	  | 2024-11-20     |Data Analyst  |			  Analyst |

Reviewer: ```suggestion
Output

| employee_id | promotion_date | new_position | previous_position |
| ----------- | -------------- | ------------ | ----------------- |
| 1      	  | 2020-01-01 	   | Junior Dev   |              NULL |
| 1      	  | 2021-06-01 	   |    Mid Dev	  | 	   Junior Dev |
| 1      	  | 2024-03-01 	   | Senior Dev   |		      Mid Dev |
| 2      	  | 2019-05-01     |     Intern	  |				 NULL |
| 2      	  | 2022-11-01 	   |    Analyst   |			   Intern |
| 2      	  | 2024-11-20     |Data Analyst  |			  Analyst |
```
Remove the word output; instead, you can use this sentence "The above code generates the below output:"
Also, please format the table properly.

| jenna      | black     | 35  | 60           |
| chris      | smith     | 30  | 35           |
| dave       | james     | 19  | 30           |

> ## Example 

Reviewer 2: We don't need to repeat the 'Example' header here.

```suggestion
```

> ### Using partition by

Reviewer 2: The header for the second example, "Example Using partition by," uses inconsistent casing and lacks proper capitalization for `PARTITION BY`.

```suggestion
### Using `PARTITION BY` Clause
```

> This example demonstrates how to use the `LAG()` function to create a new column, `previous_position`. The function retrieves the value from the previous row within the same `employee_id` group in the `Promotions` table. This is achieved by using the `PARTITION BY employee_id` clause.

Reviewer 2: The explanation for the `previous_position` column in the second example could confuse readers unfamiliar with SQL's `PARTITION BY`. It can be better explained as follows:

```suggestion
This example demonstrates how to use the `LAG()` function to create a new column, `previous_position`. 

The `PARTITION BY employee_id` clause ensures that the `LAG()` function operates within each group of rows that share the same `employee_id`. The `ORDER BY promotion_date` ensures the rows are processed in chronological order.
```

`Promotions` Table

| employee_id | promotion_date | new_position |
| ----------- | -------------- | ------------ |
| 1           | 2020-01-01     | Junior Dev   |
| 1           | 2021-06-01     |    Mid Dev   |
| 1           | 2024-03-01     | Senior Dev   |
| 2           | 2019-05-01     |     Intern   |
| 2           | 2022-11-01     |    Analyst   |
| 2           | 2024-11-20     |Data Analyst  |

```sql
SELECT *,
    LAG(new_position) OVER (
        PARTITION BY employee_id
        ORDER BY promotion_date
    ) AS previous_position
FROM Promotions;
```

> The output is a table that features a new column `previous_position`, which holds the values by employee. The first record is null because a default was not specified and the previous row would be out of range.

Reviewer 2: The explanation in the second example ("The first record is null because a default was not specified...") is almost identical to the explanation in the first example. We should rewrite it as follows:

```suggestion
Within each group defined by `employee_id`, the `previous_position` column holds the value from the previous row based on `promotion_date`. The first record in each group is `NULL` because there is no preceding row.
```

> The above code generates the below output:

Reviewer: ```suggestion
The above code generates the following output:
```
Avoid using the directional words.
